meeseeks/commands/base.py

```diff
@@ -45,99 +51,105 @@ def _get_command_methods(self):
         return command_methods
 
     @staticmethod
-    def _normalize_msg(message):
+    def _normalize_msg(message: str) -> str:
         """Return normalize message. """
 
         return ' '.join(list(map(lambda word: word.strip().lower(), message.split())))
 
-    def _get_arguments(self, command_name):
+    def _get_arguments(self, command_name: str) -> list:
         """Return arguments from the message sent by user. """
 
-        response_msg = self._normalize_msg(self._get_response_msg())
-        arguments = response_msg.replace(command_name, '', 1).strip().split(',')
-        arguments = list(map(lambda arg: arg.strip(), arguments))
+        response_msg: str = self._normalize_msg(self._get_response_msg())
+        arguments: list = response_msg.replace(command_name, '', 1).strip().split(',')
+        strip_arg: Callable[[str], str] = lambda arg: arg.strip()
+        arguments = list(map(strip_arg, arguments))
 
         return arguments if arguments[0] != '' else []
 
-    async def check_permissions(self, roles=None):
+    async def check_permissions(self, roles: Optional[list] = None) -> bool:
         """Checks if user who called the command has necessary role. """
 
-        users_info = await self._restapi.get_user_info(self._ctx.user.id)
-        allow = False
-        if not roles:
-            allow = True
-        elif any(role in roles for role in users_info['roles']):
-            allow = True
+        allow: bool = False
+        if isinstance(self._ctx, ChangedRoomMessageCtx):
+            users_info: UserInfo | None = await self._restapi.get_user_info(self._ctx.user.id)
+            if not roles:
+                allow = True
```

Comment:
Why we allow if there are no roles?

Reply:
If there are no roles, we allow execution for all users. If we check the role it means the user must have current roles